src/cpp/rtps/transport/TCPChannelResource.cpp

```diff
@@ -295,6 +295,28 @@ bool TCPChannelResource::remove_logical_port(
     return true;
 }
 
+bool TCPChannelResource::check_socket_send_buffer(
+        const size_t& msg_size,
+        const asio::ip::tcp::socket::native_handle_type& socket_native_handle)
+{
+    int bytesInSendQueue = 0;
+
+#ifndef _WIN32
+    if (ioctl(socket_native_handle, TIOCOUTQ, &bytesInSendQueue) == -1)
+    {
+        return false;
```

Comment:
If the ioctl call fails, it may mean that `TIOCOUTQ` is not supported, so I think it is better to behave as in the windows case:

```suggestion
        bytesInSendQueue = 0;
```